### PR TITLE
refactor: es10 features

### PR DESCRIPTION
**Pull request by VladStepanov**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -26,16 +18,18 @@ class VueBreadcrumbs implements PluginObject<ComponentOptions<Vue>> {
             .flatMap((route: RouteRecord) => {
               let routeRecord: RouteRecord[] = [];
 
-              if (route.meta && route.meta.breadcrumb && route.meta.breadcrumb.parent) {
+              if (route.meta?.breadcrumb?.parent) {
                 const matched = this.$router.resolve({name: route.meta.breadcrumb.parent}).route.matched
 
-                routeRecord = [...matched, ...routeRecord];
+                routeRecord = [...matched];
```
Comment (Scrum) [inline src/index.ts:24]: You can initially add `route` at `routeRecord`
```diff
- let routeRecord: RouteRecord[] = [];
+ let routeRecord: RouteRecord[] = [route];
```
and add the found here already

```diff
- routeRecord = [...matched];
+ routeRecord = [...matched, ...routeRecord];
```

while retaining the contract

```diff
- return [...routeRecord, route]
+ return routeRecord
```

File: src/index.ts [hunk 2]
```diff
@@ -26,16 +18,18 @@ class VueBreadcrumbs implements PluginObject<ComponentOptions<Vue>> {
...
               }
 
-              routeRecord.push(route)
-              return routeRecord
+              return [...routeRecord, route]
             })
-            .map(resolvePath);
+            .map((route: RouteRecord) => route.path.length === 0
```
Comment (Scrum) [inline src/index.ts:29]: It seems to me that this construction is redundant for the sake of changing one value in the object; here we can leave the construction from the function `resolvePath` ?

Comment (VladStepanov) [reply src/index.ts:29]: This construction is the same `resolvePath` function.
It seemed to me that it is redundant to put in separate function.
As you wish

Comment (VladStepanov) [reply src/index.ts:29]: It is normal practice to change the property immutably. 
Performance won't suffer (:

Comment (Scrum) [reply src/index.ts:29]: well, let it be so :)

File: src/index.ts (hunk 1 above)
Comment (Scrum) [reply src/index.ts:24]: this not changed

**Comment by Scrum** [comment]
@VladStepanov Thank you for your contribution

**Pull request merged**